src/all/foamgirl/src/eu/kanade/tachiyomi/extension/all/foamgirl/FoamGirl.kt

```diff
@@ -0,0 +1,128 @@
+package eu.kanade.tachiyomi.extension.all.foamgirl
+
+import android.util.Log
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class FoamGirl() : ParsedHttpSource() {
+    override val baseUrl = "https://foamgirl.net"
+    override val lang = "all"
+    override val name = "FoamGirl"
+    override val supportsLatest = false
+
+    override fun headersBuilder() = Headers.Builder().apply {
+        add("Referer", baseUrl)
+    }
+
+    // Popular
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        thumbnail_url = element.select("img").attr("data-original")
+        val page_count = element.select(".postlist-imagenum").text()
+        title = element.select("a.meta-title").text()
+        setUrlWithoutDomain(element.select("a").attr("href"))
+        initialized = true
+    }
+
+    override fun popularMangaNextPageSelector() = "a.next"
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/page/$page", headers)
+    }
+
+    override fun popularMangaSelector() = ".update_area .i_list"
+
+    // Search
+
+    override fun searchMangaFromElement(element: Element) = popularMangaFromElement(element)
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        return GET(
+            baseUrl.toHttpUrl().newBuilder().apply {
+                addPathSegment("page")
+                addPathSegment("$page")
+                addQueryParameter("post_type", "post")
+                addQueryParameter("s", query)
+            }.build(),
+            headers,
+        )
+    }
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    // Details
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        title = document.select("h1").text()
```

Comment:
nothing else available to perse/update? if so override `fetchMangaDetails` to return same copy of manga. since updating title alone isn't needed.